apps/ntag/arduinoNtag/ArduinoWire.cpp

```diff
@@ -0,0 +1,278 @@
+/*
+ * ArduinoWire.cpp
+ *
+ * Implementation of the ArduinoWire class
+ *
+ *  Created on: Apr 4, 2016
+ *      Author: mario
+ *	Modified (mostly rewrote): SWI for mangOH Yellow
+ */
+
+#include <sys/ioctl.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <linux/types.h>
+#include <linux/i2c.h>
+#include <linux/i2c-dev.h>
+#include <iostream>
+#include <stdint.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include "legato.h"
+#include "ArduinoWire.h"
+#include "ntag.h"
+#include "NDEF/Ndef.h"
+
+ArduinoWire::ArduinoWire(): fd(0), txAddress(0), txBufferIndex(0), txBufferLength(0), rxBufferIndex(0), rxBufferLength(0), releaseBus(true) {
+
+	/* Let's see what functionality our I2c adapter has
+	unsigned long funcs; */
+
+	// TODO: Parameterize. On a mangOH Yellow the ntag chip is on i2c bus 8
+        if(access("/dev/i2c-8", O_RDWR) == -1) {
+                LE_INFO("Cannot access /dev/i2c-8 for O_RDWR error: %s", strerror(errno));
+                exit(1);
+        }
+        /* else
+                LE_INFO("Can access /dev/i2c-8"); */
+
+	this->fd = open("/dev/i2c-8", O_RDWR);
+
+	if (this->fd == -1) {
+                LE_INFO("Cannot open /dev/i2c-8 O_RDWR ret: %d %s", this->fd, strerror(errno));
+                exit(2);
+	}
+
+	/* if(ioctl(this->fd, I2C_FUNCS, &funcs) < 0) {
+		LE_INFO("I2C_FUNCS returned error");
+		exit(3);
+  	}
+  	LE_INFO("I2c Funcs of adapter: %ld", funcs); */
+}
+
+void ArduinoWire::becomeBusMaster() {
+	if (txAddress == 0) {
+		LE_INFO("Could not communicate with slave device at address: %x", txAddress);
+		exit(1);
+	}
+	/* SWI: We aren't doing SMBus transactions we are using the combined interface
+	if (ioctl(this->fd, I2C_SLAVE_FORCE, txAddress) < 0) {
+		LE_INFO("Could not I2C_SLAVE_FORCE slave device at address: %x", txAddress);
+		exit(1);
+	}
+	*/
+
+}
+void ArduinoWire::begin(){
+	txBufferIndex = 0;
+	txBufferLength = 0;
+
+	rxBufferIndex = 0;
+	rxBufferLength = 0;
+}
+
+//void ArduinoWire::begin(uint8_t address) {
+//	txAddress = address;
+//
+//	begin();
+//}
+
+void ArduinoWire::beginTransmission(const uint8_t address){
+	/* SWI: We aren't doing SMBus transactions we are using the combined interface
+	int ret;
+        if ((ret = ioctl(fd, I2C_SLAVE_FORCE, address)) < 0) {
```

Comment:
There's lots of dead code in this file.  This should be removed now that you have settled on an implementation.